### DISCUSSION TITLE
Novelty detection - semi-supervised learning

### DISCUSSION THREAD

**Discussion by aaronfderybel** [yzhao062/pyod#480] [General]
Hello PyoD community,

I was wondering what the recommended way to perform Novelty detection is using this library.
According to https://pyod.readthedocs.io/en/latest/relevant_knowledge.html novelty detection is the situation where you have access to a dataset with only inliers. You train your algorithm on this dataset. Any new data could be inlier or outliers.

Would the recommended way to use these algorithms for novelty detection:
1. Train on dataset with only inlier labels
2. predict on unseen dataset with mixed inlier/outlier
3. Calculate scores on unseen data if you have labels.

Are there any algorithms that are more suitable for novelty detection than others?

**Comment by John-Almardeny** [comment]
I would go with the `AutoEncoder` for this purpose if you want to train only on the inliers because you can calculate a threshold for inliers then any point that exceeds this threshold can be a novel one!
